pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -119,11 +125,31 @@ func (configgen *ConfigGeneratorImpl) buildOutboundClusters(cb *ClusterBuilder,
 			if port.Protocol == protocol.UDP {
 				continue
 			}
-			lbEndpoints := cb.buildLocalityLbEndpoints(networkView, service, port.Port, nil)
+			clusterName := model.BuildSubsetKey(model.TrafficDirectionOutbound, "", service.Hostname, port.Port)
+			clusterKey := &clusterCache{
+				clusterName:     clusterName,
+				service:         service,
+				networkView:     networkView,
+				pushVersion:     cb.push.PushVersion,
+				destinationRule: cb.push.DestinationRule(cb.proxy, service),
+				locality:        cb.proxy.Locality,
+				proxySidecar:    cb.proxy.Type == model.SidecarProxy,
+				http2:           port.Protocol.IsHTTP2(),
+				downstreamAuto:  cb.proxy.Type == model.SidecarProxy && util.IsProtocolSniffingEnabledForOutboundPort(port),
+			}
+			cached, tokens, allFound := cb.getAllCachedSubsetClusters(*clusterKey)
+			if allFound && !features.EnableUnsafeAssertions {
+				hit += len(cached)
```

Comment:
Because of how tightly coupled the default cluster is with its subsets, this feels more like we are scaling the hit/miss rates by the "size" of the set of clusters. 

Not sure if it actually makes more sense to count the "set" of clusters as defined by host/port though..

Reply:
I see what you mean, one thing I like about this is when I fix the logging to properly include the caching we will see something like `CDS: push resources:18 cached:7/18`. If we do it by "set", we may see  `CDS: push resources:18 cached:3/7` or something, which may be a bit confusing?

Not a strong preference though